ThisSensorManager-back.php

```diff
@@ -0,0 +1,469 @@
+            // dump($state);
+            // dump($state);
+            // echo array_key_first($ids[$id]);
+            //dump( $state[key($state)]);
+            //echo "ID: ".key($state)."STATE ->".$state[key($state)];
+            switch ($sState) {
+                case 0:
+                    if($state[key($state)] == self::STOP)
+                    {
+                        $sState++;
+                        $start = $this->database->table($sName)->where("id=?", array_key_first($ids[$id]))->fetch()->time->getTimestamp();
+                        //echo " -> STOP -> ".$start;
+                    }                    
+                    break;
+                case 1:
+                    if($state[key($state)] == self::REWORK)
+                    {
+                        $stop = $this->database->table($sName)->where("id=?", array_key_first($ids[$id]))->fetch()->time->getTimestamp();
+                        $sState = 0;
+                        $time += $stop-$start;
+                        //echo " -> REWORK -> ".$stop."-> ALL: ". $time;
+                    }
+                    break;
+
+                // default:
+                //     # code...
+                //     break;
+            }
+            //echo "<br><br>";
+        }        
+
+        // $first = $this->database->table($sName)->where("id=?", $ids[0])->fetch()->time->getTimestamp();
+        // $last = $this->database->table($sName)->where("id=?", end($ids))->fetch()->time->getTimestamp();
+        // $res =  $last-$first;
+        return array(new DateInterval("PT".$time."S"), $time);
+
+    }
+
+    public function getAproxStopTime($stopTimestamp, $stopCount)
+    {
+        $time = ceil($stopTimestamp/$stopCount);
+        return array(new DateInterval("PT".$time."S"), $time);
+    }    
+
+    public function getWorkTime($allTimestamp, $stopTimestamp)
+    {
+        $time = $allTimestamp-$stopTimestamp;
+        return array(new DateInterval("PT".$time."S"), $time);
+    }
+
+
+
+    public function getAproxWorkTime($workTimestamp, $workCount)
+    {
+        $time = ceil($workTimestamp/$workCount);
+        return array(new DateInterval("PT".$time."S"), $time);
+    }
+
+    public function getWorkTimeOld($sName, $ids): DateInterval
+    {
+        $last = $this->database->table($sName)->where("id=?", 1)->fetch();
+        //echo $out =  $last->time->getTimestamp();
+        //dump($last->time);
+        $workTime = 0;
+
+        // $next = $this->database->table($sName)->where("id=?", 2)->fetch();
+        // //echo $next->time->getTimestamp();
+        // dump($next->time);
+
+
+        // $start = DateTime::from($out);
+
+        // //echo $start;
+
+
+        foreach($ids as $id)
+        {
+            $actual = $this->database->table($sName)->where("id=?", $id)->fetch();
+
+            //dump($last->actWork);
+
+            //  First loop
+            if($id < $ids[1])
+            {
+                $actWork=3*self::MINUTE; // 3 Minutes
```

Comment:
`actWork`?

I would prefer not to shorten the name: `actualWork` as you have IntelliSense which will help you. 

But still what is the meaning of `actualWork`. Wouldn't it be better to have `workingCumulativeTime`?

Reply:
Ok `workingCumulativeTime` is not a good name. What is the purpose of this variable?